internal/pr/promote/github_pull_request_provider.go

```diff
@@ -14,30 +14,10 @@ type GitHubPullRequestProvider struct {
 var labelRegex = regexp.MustCompile(`^promote:\s*(\w+)$`)
 
 type pullRequest struct {
-	Labels []struct {
+	HeadRefName string `json:"headRefName"`
+	Labels      []struct {
```

Comment:
I am wondering if we should use the proper github go library even if we just stole the `gh` auth token from the keychain? There is already a fair amount of dancing around (parsing stdout) of the output of `gh`.

![image](https://github.com/nestoca/joy/assets/4259753/466951c8-56ff-4b52-948b-aee0f2bcfc9d)


https://github.com/cli/cli/blob/trunk/internal/keyring/keyring.go#L34C7-L53

Reply:
That would be great, I initially was hoping to do that, but couldn't figure out how to leverage the user's current authorization via API. However, please create another ticket for that one, as we really need to conclude this one! 😁